src/interoperability.md

```diff
@@ -272,7 +272,27 @@ be lowercase without trailing punctuation, and typically concise.
 [`Error::description()`] should not be implemented. It has been deprecated and users should
 always use `Display` instead of `description()` to print the error.
 
+If an error type returns an underlying source error from [`Error::source()`], it does **not**
+include that source error in its own `Display` representation as well. This avoids duplicated
+information when an error is printed along with all its sources:
+
+```rust
+use std::fmt;
+
+impl fmt::Display for ParseError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // Good errors never duplicate information like this:
```

Comment:
i'm aware that the existing example already does the good/bad thing, but it might be better to just show the "good" thing? but then make it a full-fledged example (i.e. also include the `impl std::error::Error for ParseError` with `fn source` implemented?